$NetBSD: patch-app_Main.lhs,v 1.2 2025/08/20 19:23:36 wiz Exp $

Workaround for happy chokes on non-ASCII inputs when the locale isn't
"*.UTF-8". This is probably not a desirable way to fix it but a proper fix
would need a serious work. Without this workaround we cannot build
devel/hs-ghc-lib-parser.

TODO: Report this to upstream and let them decide what to do.

--- app/Main.lhs.orig	2001-09-09 01:46:40.000000000 +0000
+++ app/Main.lhs
@@ -29,12 +29,14 @@ Path settings auto-generated by Cabal:
 > import System.Exit (exitWith, ExitCode(..))
 > import Data.Char
 > import Data.List ( union )
+> import GHC.IO.Encoding (setLocaleEncoding)
 > import System.IO
 > import Data.List( isSuffixOf )
 > import Data.Version ( showVersion )
 
 > main :: IO ()
 > main = do
+>       setLocaleEncoding utf8
 
 Read and parse the CLI arguments.
 
